pandas/tests/arrays/test_integer.py

```diff
@@ -659,6 +659,7 @@ def test_cross_type_arithmetic():
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.filterwarnings("ignore:NDFrame.to_dense:FutureWarning")
```

Comment:
I don't want to ignore these; instead if we are actually warning, then we need to adjust the internals to not do this (or *maybe* just catch the warning); these are essentially user facing tests

Reply:
this comments applies to virtually all of the remaining filter warnings)

Reply:
Not quite sure what you mean by user facing tests. This only ignores the warning during CI tests for pandas and not anything built using pandas. 

Also, I have written separate tests which specifically checks for and catches these warnings [here](https://github.com/pandas-dev/pandas/pull/26684/commits/50d0534f8c024686e2f939718c7e7e1cb19ab441#diff-227fb1652ca8fc0912cd32ed1d536297), so in case something goes wrong there, those tests would fail.

Reply:
we do want to catch these warnings at all, rather to fix where they are coming from.

*only* a routine that is explicity calling ``.to_dense()`` should w catch, none of these tests do that. We are calling ``.to_dense()`` somewhere in the implementation; that needs to be fixed.

Reply:
Those integer array groupby tests should indeed not generate a warning. @intEll1gent can you show the warning traceback that you get for the code of this test?

One tip can be to elevate the warning to an error to debug the location where it is coming from.

Reply:
The warning comes from usage of `to_dense` in `pandas/core/groupby/ops.py` lines 633, 635, 882. In fact, this seems to be the case for many of the other `NDFrame.to_dense` deprecation warnings. Removing `.to_dense` should do the job since the method simply returns whats passed in.

Reply:
You might need to be careful about simply removing `to_dense`, because if you have sparse data in the DataFrame, then it actually does something.

Reply:
The methods I was thinking of changing seem to be operating on series. This is one of those three:
`class SeriesSplitter(DataSplitter):`
` def _chop(self, sdata, slice_obj):`
`return sdata._get_values(slice_obj).to_dense()`

I was thinking about replacing it with 
`class SeriesSplitter(DataSplitter):`
       `def _chop(self, sdata, slice_obj):`
            `return sdata._get_values(slice_obj)`

Local tests all pass with these changes.

Reply:
@jorisvandenbossche Are the changes I suggested in my previous comment fine or would it do something we don't want it to?

Reply:
That's what I understood you proposed. If the tests pass, that is probably fine, so I would certainly push it (I was just wondering if one does a groupby with sparse columns, if the to_dense there would be needed, but then probably not)